obspy/clients/eida/fetch.py

```
@@ -0,0 +1,950 @@
+                            end_time.isoformat())
+
+    def parse(self, path):
+        with open(path) as fd:
+            for line in fd:
+                line = line.rstrip()
+
+                if line:
+                    self.__parse_line(line)
+
+
+class BreqParser(object):
+    __tokenrule = "^\.[A-Z_]+[:]?\s"
+
+    __reqlist = ("(?P<station>[\w?\*]+)",
+                 "(?P<network>[\w?]+)",
+                 "((?P<beg_2year>\d{2})|(?P<beg_4year>\d{4}))",
+                 "(?P<beg_month>\d{1,2})",
+                 "(?P<beg_day>\d{1,2})",
+                 "(?P<beg_hour>\d{1,2})",
+                 "(?P<beg_min>\d{1,2})",
+                 "(?P<beg_sec>\d{1,2})(\.\d*)?",
+                 "((?P<end_2year>\d{2})|(?P<end_4year>\d{4}))",
+                 "(?P<end_month>\d{1,2})",
+                 "(?P<end_day>\d{1,2})",
+                 "(?P<end_hour>\d{1,2})",
+                 "(?P<end_min>\d{1,2})",
+                 "(?P<end_sec>\d{1,2})(\.\d*)?",
+                 "(?P<cha_num>\d+)",
+                 "(?P<cha_list>[\w?\s*]+)")
+
+    def __init__(self):
+        self.__rx_tokenrule = re.compile(BreqParser.__tokenrule)
+        self.__rx_reqlist = re.compile("\s+".join(BreqParser.__reqlist))
+        self.postdata = ""
+        self.failstr = ""
+
+    def __parse_line(self, line):
+        m = self.__rx_reqlist.match(line)
+
+        if m:
+            d = m.groupdict()
+
+            # catch two digit year inputs
+            if d["beg_2year"]:
+                if int(d["beg_2year"]) > 50:
+                    d["beg_4year"] = "19%s" % d["beg_2year"]
+
+                else:
+                    d["beg_4year"] = "20%s" % d["beg_2year"]
+
+            if d["end_2year"]:
+                if int(d["end_2year"]) > 50:
+                    d["end_4year"] = "19%s" % d["end_2year"]
+
+                else:
+                    d["end_4year"] = "20%s" % d["end_2year"]
+
+            # some users have problems with time...
```

IMHO better to just raise an error and force users to fix their files.